www/writeLoginApi.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+declare(strict_types=1);
+
+use SimpleSAML\Logger;
+use SimpleSAML\Module\proxystatistics\Config;
+use SimpleSAML\Module\proxystatistics\DatabaseCommand;
+
+if ('POST' !== $_SERVER['REQUEST_METHOD'] && 'PUT' !== $_SERVER['REQUEST_METHOD']) {
+    Logger::info(
+        'proxystatistics:writeLoginApi - API write called not using POST nor PUT, returning 405 response code'
+    );
+    header('HTTP/1.0 405 Method Not Allowed');
+    exit;
+}
+$config = Config::getInstance();
+
+if (!$config->isApiWriteEnabled()) {
+    Logger::info(
+        'proxystatistics:writeLoginApi - API write called, but disabled in config. Returning 501 response code'
+    );
+    header('HTTP/1.0 501 Not Implemented');
+    echo '<p>Access denied. You did not enter a password.</p>';
+    exit;
+}
+
+$authUsername = $_SERVER['PHP_AUTH_USER'] ?? '';
+$authPass = $_SERVER['PHP_AUTH_PW'] ?? '';
+
+$username = $config->getApiWriteUsername();
+$password = $config->getApiWritePassword();
+
+// If we get here, username was provided. Check password.
+if ($authUsername !== $username || $authPass !== $password) {
+    Logger::info(
+        'proxystatistics:writeLoginApi - API write called with bad credentials (' . $authUsername . ':' . $authPass . ') returning 401 response code'
+    );
+    header('HTTP/1.0 401 Unauthorized');
+    exit;
+}
+
+$data = json_decode(file_get_contents('php://input'), true);
+
+$userId = $data[DatabaseCommand::API_USER_ID] ?? null;
+$serviceIdentifier = $data[DatabaseCommand::API_SERVICE_IDENTIFIER] ?? null;
+$serviceName = $data[DatabaseCommand::API_SERVICE_NAME] ?? null;
+$idpIdentifier = $data[DatabaseCommand::API_IDP_IDENTIFIER] ?? null;
+$idpName = $data[DatabaseCommand::API_IDP_NAME] ?? null;
+
+if (null === $userId || null === $serviceIdentifier || null === $serviceName || null === $idpIdentifier || null === $idpName) {
```

Comment:
```suggestion
if (!empty(array_diff([DatabaseCommand::API_USER_ID, DatabaseCommand::API_SERVICE_IDENTIFIER, DatabaseCommand::API_SERVICE_NAME, DatabaseCommand::API_IDP_IDENTIFIER, DatabaseCommand::API_IDP_NAME], array_keys($data)))) {
```
The content of the variables is not used anyway.